lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -86,11 +86,10 @@ defmodule Mix.Compilers.Elixir do
         compile_manifest(manifest, exts, modules, sources, stale, dest, timestamp, opts)
       removed != [] ->
         write_manifest(manifest, modules, sources, dest, timestamp)
+        {:ok, warning_diagnostics(sources)}
```

Comment:
I assume we want this because of the --all-warnings flag?

Reply:
Yeah, the idea is that if we're displaying these warnings inline in an editor, we don't want warnings to disappear just because the file hasn't changed since the last compile, especially since compilation might be automatic when files are saved.